packages/xrpl/src/client/index.ts

```diff
@@ -666,8 +667,39 @@ class Client extends EventEmitter<EventTypes> {
       promises.push(checkAccountDeleteBlockers(this, tx))
     }
 
-    return Promise.all(promises).then(() => tx)
+    // further manipulation of tx_ uses non-SubmittableTransaction types, hence we need a typecast to any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/consistent-type-assertions -- `any` type is required to perform non-protocol modifications to the JSON object
+    const tx_ = tx as any
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A transaction JSON must contain TransactionType field
+    if (tx_.TransactionType === 'Payment' && tx_.DeliverMax != null) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A Payment transaction JSON must contain Amount field
+      if (tx_.Amount == null) {
```

Comment:
```suggestion
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A transaction JSON must contain TransactionType field
    if (tx.TransactionType === 'Payment' && tx.DeliverMax != null) {
      const paymentTx = tx as Payment
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A Payment transaction JSON must contain Amount field
      if (paymentTx.Amount == null) {
```
It's better to do type assertion when we know `tx` is a `Payment` and we can avoid using `any`.

Reply:
We need to deal with the `DeliverMax` field until line 696 (`delete tx_.DeliverMax`). This explicit type cast can be performed after this line, otherwise the compiler will complain: `Property 'DeliverMax' does not exist on type 'Payment'. Did you mean 'DeliverMin'?`

Reply:
updated in [60f3e6f](https://github.com/XRPLF/xrpl.js/pull/2689/commits/60f3e6fecc11db6d846b348731b29c38059796b0), thanks for the help @khancode